### PR TITLE
solution

### PR DESCRIPTION
**Pull request by RogerHuba**
- [x] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
     - [x] “Happy Path” - Expected outcome
     - [x] Expected failure
     - [x] Edge Case (if applicable/obvious)
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [x] Link to code

### CODE REVIEW

File: 401d56/code-challenges/gradle/src/main/java/hashtable/Node.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    public void setKey(String key){
+        this.key = key;
+    }
+
+    public void setValue(String value) {
+        this.value = value;
+    }
+
+    public void Node(String key, String value){
```
Comment (VorosDana) [inline 401d56/code-challenges/gradle/src/main/java/hashtable/Node.java]: This is named the same as the class, but isn't the constructor. Why? myNode.Node(key, value) doesn't tell me what it's doing at all

File: 401d56/code-challenges/gradle/src/main/java/hashtable/Hashtable.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    // getLength
+    public int getLength(){
+        int value = map.length;
+        return value;
+    }
+
+    // contains
+    public Boolean contains(String key){
```
Comment (VorosDana) [inline 401d56/code-challenges/gradle/src/main/java/hashtable/Hashtable.java]: A bucket can have multiple things in it, it's why we use things like linked lists to hold the items in those buckets. This will return true if a key would go to a bucket that already has something in it, but isn't there. You need to check through the nodes in the bucket

File: 401d56/code-challenges/gradle/src/main/java/hashtable/Hashtable.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        hashValue = (hashValue * 843) % map.length;
+        return hashValue;
+    }
+
+    public void set(String key, String value) {
+        int hashkey = hash(key);
+
+        if (this.map[hashkey] == null) {
+            map[hashkey] = new Node(key, value);
+        } else {
```
Comment (VorosDana) [inline 401d56/code-challenges/gradle/src/main/java/hashtable/Hashtable.java]: What should be done if a key already exists in the map? You add it to the list, but it can't ever be found, you return the first node with the key you're looking for
